drivers/ethernet/eth_w5500.c

```diff
@@ -340,6 +340,17 @@ static int w5500_set_config(const struct device *dev,
 			    enum ethernet_config_type type,
 			    const struct ethernet_config *config)
 {
+	struct w5500_runtime *ctx = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
```

Comment:
Thanks for changes. Could you please adapt for `ETHERNET_CONFIG_TYPE_PROMISC_MODE` as well with the case statement?